### DISCUSSION TITLE
how can i load 3d model from internal storage

### DISCUSSION THREAD

**Discussion by mehdiasgari** [sceneview/sceneview#583] [Q&A]
i want to use an internal storage address to load my Model instead of use the assets folder address. can you help me?

**Comment by mehdiasgari** [comment]
val fileName = "a4.glb"
            val file = File(context.filesDir, fileName)
            val address :String = file.absolutePath


childNodes = listOf(centerNode,
                rememberNode {
                    ModelNode(
                        modelInstance = modelLoader.createModelInstance(
                            assetFileLocation = address ),
                        scaleToUnits = 1.5f
                    )
this is my code. but it does not work correctly!
this the my application opens the model's folder and does not open the model

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, `rememberModelInstance` loads from assets by default. For internal storage or arbitrary file paths, use `modelLoader` directly:

```kotlin
SceneView {
    val model = rememberModelInstance(
        modelLoader = modelLoader,
        fileLocation = "file:///path/to/model.glb"
    )
    if (model != null) {
        ModelNode(modelInstance = model)
    }
}
```

You can also load from a URL or content URI. The loader supports `file://`, `content://`, and `https://` schemes in addition to asset paths.
